### ISSUE TITLE
Questions Regarding Suboptimal Performance of RadarGNN Reproduction

### ISSUE THREAD

**Issue by BillionLee** [TUMFTM/RadarGNN#8]
Below are the details of my environment and the steps I followed:

Environment Details (inside the Docker container):

Operating System:

cat /etc/os-release
# Output: Ubuntu 18.04.6 LTS (Bionic Beaver)
(Note that the repository recommends Ubuntu 20.04 LTS.)

NVIDIA Driver and CUDA:

nvidia-smi
# Output shows Driver Version: 565.57.01 and CUDA Version: 12.7
PyTorch:

python3 -c "import torch; print('PyTorch version:', torch.__version__)"
# Output: PyTorch version: 1.11.0
CUDA Version (as detected by PyTorch):

python3 -c "import torch; print('CUDA version (PyTorch):', torch.version.cuda)"
# Output: CUDA version (PyTorch): 11.3
cuDNN Version:

python3 -c "import torch; print('cuDNN version:', torch.backends.cudnn.version())"
# Output: cuDNN version: 8200 (i.e., cuDNN 8.2)
Python Version: 3.8.12

Installed Key Dependencies:

torch-scatter==2.0.9

torch-sparse==0.6.15

torch-cluster==1.6.0

torch-spline-conv==1.2.1

torch-geometric==2.1.0.post1

Commands Executed:

Dataset Creation:

python3 src/gnnradarobjectdetection/create_dataset.py --dataset data/datasets/radarscenes --config configurations/configuration_radarscenes.yml
Training:

python3 src/gnnradarobjectdetection/train.py --data data/datasets/radarscenes/processed --results data/results --config configurations/configuration_radarscenes.yml
Evaluation:
After modifying the model-loading section to load the desired model (e.g., using trained_model_low_val_ep15.pt), I executed:

python3 src/gnnradarobjectdetection/evaluate.py --data data/datasets/radarscenes/processed --model data/results/model_01 --config configurations/configuration_radarscenes.yml
Evaluation Results:
Using trained_model_low_val_ep15.pt, the evaluation result was:

{
    "OBJECT_DETECTION_METRICS": {
        "mAP": 0.5447449088096619,
        "mAP_per_class": [
            0.727835476398468,
            0.17905427515506744,
            0.5689184069633484,
            0.5972417593002319,
            0.6506749987602234
        ]
    },
    "SEMANTIC_SEGMENTATION_METRICS": {
        "f1": [
            0.7755512200521645,
            0.4452016002157594,
            0.7926296602456955,
            0.8152789005658853,
            0.7421696855694867,
            0.9930660054231483
        ]
    }
}
Modifying the model file to load trained_model.pt yielded a slightly higher mAP of approximately 0.5749, which is still below the results reported in your paper.

My Questions/Concerns:

OS Version Difference:
My Docker container is based on Ubuntu 18.04 LTS, whereas the recommended environment specifies Ubuntu 20.04 LTS. Could this version difference influence the results? If so, do you recommend running on Ubuntu 20.04?

Data Preprocessing & Configuration:
Could there be subtle differences in dataset preprocessing, configurations (e.g., crop settings, bounding box invariance parameters), or training hyperparameters (learning rate, epochs, batch size) that might explain the lower performance? Are there any additional details or adjustments to these settings that are critical for achieving the reported results?

Dependency Versions:
I verified that my installation of torch-geometric and other libraries (torch-scatter, torch-sparse, torch-cluster, torch-spline-conv) is consistent with the requirements. Are there any known issues or recommended patch versions for any of these libraries that might affect the final performance?

I would greatly appreciate any insights or suggestions you might provide to help me bridge this performance gap.

Thank you very much for your time and assistance.
